src/test/txvalidationcache_tests.cpp

```diff
@@ -109,10 +109,15 @@ BOOST_FIXTURE_TEST_CASE(tx_mempool_block_doublespend, TestChain100Setup)
 static void ValidateCheckInputsForAllFlags(const CTransaction &tx, uint32_t failing_flags, bool add_to_cache) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
 {
     PrecomputedTransactionData txdata;
-    // If we add many more flags, this loop can get too expensive, but we can
-    // rewrite in the future to randomly pick a set of flags to evaluate.
-    for (uint32_t test_flags=0; test_flags < (1U << 16); test_flags += 1) {
+
+    FastRandomContext insecure_rand(true);
+
+    for (int count = 0; count < 10000; ++count) {
         TxValidationState state;
+
+        // Randomly selects flag combinations
+        uint32_t test_flags = (uint32_t) insecure_rand.randrange((SCRIPT_VERIFY_END_MARKER - 1) << 1);
```

Comment:
You want to swap the - and <<; right now you get `111...0`, i.e.
```suggestion
        uint32_t test_flags = (uint32_t) insecure_rand.randrange((SCRIPT_VERIFY_END_MARKER << 1) - 1);
```

Reply:
Not really.
Right now the last flag is 1<<16.
Bc of the way the enum works my new constant becomes (1<<16)+1. That’s why I used the “...PLUS_ONE” constant before.
So the argument to the range is (((1<<16)+1)-1)<<1 = (1<<16)<<1 = 1<<17
Because the range is exclusive, the highest possible value is (1<<17)-1 = 1111111111111111b

Reply:
Ahh, I misread it as a mask for some reason.